### PR TITLE
Treat special case Ez=0 for ChrAcc element.

### PR DESCRIPTION
**Pull request by cemitch99**
The ChrAcc element map (uniform Ez + Bz field) requires special handling for the edge case that Ez=0.  This PR adds the necessary changes.

- [x] modify push
- [x] add a benchmark test

This partially addresses Issue #1573

### CODE REVIEW

File: tests/python/test_chracc_Ez0.py [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+            dpt,
+            qm_eev,
+            bunch_charge=0.0,
+        )
+
+    # store initial particles
+    df_initial = pc.to_df()
+
+    # design the accelerator lattice
+    ns = 50  # number of slices per ds in the element
```
Comment (cemitch99) [inline tests/python/test_chracc_Ez0.py]:
```suggestion
    ns = 1  # number of slices per ds in the element
```

File: tests/python/test_chracc_Ez0.py [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    # set numerical parameters and IO control
+    sim.space_charge = False
+    sim.slice_step_diagnostics = True
+
+    # domain decomposition & space charge mesh
+    sim.init_grids()
+
+    # basic beam parameters
+    kin_energy_MeV = 2000.0  # reference energy (kinetic)
+    bunch_charge_C = 25.0e-12  # used with space charge
```
Comment (cemitch99) [inline tests/python/test_chracc_Ez0.py]:
```suggestion
```

File: src/elements/ChrUniformAcc.H [hunk 3]
```diff
@@ -112,6 +112,9 @@ namespace impactx::elements
             // compute focusing constant (1/m) and rotation angle (in rad)
             m_alpha = m_bz * 0.5_prt;
             m_alpha_iez = m_alpha / m_ez;
+
+            // dimensionless acceleration parameter (change in gamma)
+            m_acc = m_ez * m_slice_ds;
```
Comment (ax3l) [inline src/elements/ChrUniformAcc.H:116]: Minor note (ok to ignore): `m_alpha` and `m_acc` are probably not worth caching and would be ok to recompute for every particle, but also does not hurt as we benchmarked recently.

File: src/elements/ChrUniformAcc.H
Comment (ax3l) [inline src/elements/ChrUniformAcc.H]: Probably a bug: In the reference particle, there is still a division by zero here and the zero-path is not treated.

File: tests/python/test_chracc_Ez0.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        "momentum_y",
+        "momentum_t",
+    ]
+
+    for c in PHASE_COLS:
+        np.testing.assert_allclose(
+            df_final[c].to_numpy(),
+            df_initial[c].to_numpy(),
+            atol=1.0e-12,
+            rtol=0,
```
Comment (ax3l) [inline tests/python/test_chracc_Ez0.py:167]: Here, can you also assert the reference particle state for correctness?

File: src/elements/ChrUniformAcc.H
Comment (ax3l) [inline src/elements/ChrUniformAcc.H]: Probably same bug: The transport map still has a division by zero / no treatment of the zero case.

Comment (cemitch99) [reply src/elements/ChrUniformAcc.H]: Yes, good catch.  Will fix.

File: tests/python/test_chracc_Ez0.py (hunk 4 above)
Comment (cemitch99) [reply tests/python/test_chracc_Ez0.py:167]: Will do.

File: src/elements/ChrUniformAcc.H
Comment (cemitch99) [reply src/elements/ChrUniformAcc.H]: Fixed.

**Comment by cemitch99** [comment]
@ax3l Thanks for the comments.  Suggestions addressed; ready for re-review.

File: src/elements/ChrUniformAcc.H
Comment (ax3l) [reply src/elements/ChrUniformAcc.H]: Did you mean to comment in line 270 and remove line 271? :)

Comment (ax3l) [reply src/elements/ChrUniformAcc.H]: fixed

Comment (cemitch99) [reply src/elements/ChrUniformAcc.H]: Fixed.

File: src/elements/ChrUniformAcc.H [hunk 5]
```diff
@@ -169,7 +172,7 @@
                 T_Real const denom = -pti_tot + pzi_tot;
 
                 // compute focusing constant (1/m) and rotation angle (in rad)
-                T_Real const theta = m_alpha_iez * log(numer / denom);
+                T_Real const theta = (m_acc == 0_prt)? m_alpha * m_slice_ds / pzi_tot : m_alpha_iez * log(numer / denom);
```
[the remark this answers is not in the document]
Comment (ax3l) [reply src/elements/ChrUniformAcc.H]: Ok for now. We could use a machine epsilon as in `abs(m_acc) < machine eps` to make sure simple preparatory calculations that end up close to zero are also hitting here. But for now this is ok, for the special case that is directly set.

File: src/elements/ChrUniformAcc.H [hunk 6]
```diff
@@ -169,7 +172,7 @@ namespace impactx::elements
...
```
Comment (ax3l) [inline src/elements/ChrUniformAcc.H]: Codex:

  - `Ez=0, Bz=0` still produces NaNs on this line (175-190)
    Here theta=0 and m_alpha=0, so the subsequent sin_theta / m_alpha evaluates
    as 0/0. An `alpha==0` branch or stable `sinc` formulation is needed.

File: src/elements/ChrUniformAcc.H [hunk 7]
```diff
@@ -112,6 +112,9 @@ namespace impactx::elements
             // compute focusing constant (1/m) and rotation angle (in rad)
             m_alpha = m_bz * 0.5_prt;
             m_alpha_iez = m_alpha / m_ez;
```
Comment (ax3l) [inline src/elements/ChrUniformAcc.H]: Codex (is right):

  - [P2] The zero-field path still evaluates a division by zero.
    `m_alpha_iez = m_alpha / m_ez` executes _bef_ore the new branch. It leaves an
    infinity in normal execution and aborts when the documented
    `amrex.fpe_trap_zero` debugging option is enabled. Compute it conditionally
    or restructure around the finite integrated-drift factor.

File: tests/python/test_chracc_Ez0.py [hunk 8]
```diff
@@ -0,0 +1,129 @@
...
+    sim.init_grids()
+
+    # basic beam parameters
+    kin_energy_MeV = 2000.0  # reference energy (kinetic)
+
+    # set reference particle
+    ref = sim.beam.ref
+    ref.set_species("proton").set_kin_energy_MeV(kin_energy_MeV)
+    qm_eev = ref.charge_qe / (ref.mass_MeV * 1.0e6)  # electron charge/mass in e / eV
+    initial_ref = ref
```
Comment (ax3l) [inline tests/python/test_chracc_Ez0.py]: Codex:

  - [P2] The reference-particle regression assertion is tautological —
    line 31-35

    `beam.ref` is a live mutable object, so `initial_ref = ref` aliases it. The
    final loop compares the reference particle with itself. Use `initial_ref = ref.copy()`, as the existing reversibility tests do.

**Review by ax3l — APPROVED** [review]
I was about to approve this but Codex found a few more bugs.

>  The exact Ez → 0 formulas for particle t, reference t, drift_log, and R56
  otherwise look correct. I’d also consider adding nonzero transverse
  coordinates and a linear-map check; the current test does not exercise the
  new transverse Pxy_mech2 term or transport_map.

**Comment by ax3l** [comment]
@cemitch99 I can add commits for those fixes and you can review them quickly.

File: src/elements/ChrUniformAcc.H (hunk 7 above)
Comment (cemitch99) [reply src/elements/ChrUniformAcc.H]: This quantity is not used at all when `m_ez=0`, so we can assign zero in this case.  Otherwise, it might be easier simply to remove it, and accept the division cost.

Comment (ax3l) [reply src/elements/ChrUniformAcc.H]: fixed via 3d0c69c976873b8f04d9219de94a81a662df6abf

File: tests/python/test_chracc_Ez0.py (hunk 8 above)
Comment (ax3l) [reply tests/python/test_chracc_Ez0.py]: fixed via b1ba61a5898b65777550b1cda7de474330e35d75

test expanded via 40d62d431b5a44fd871244850e8929e0b53b03d4

File: src/elements/ChrUniformAcc.H (hunk 6 above)
Comment (ax3l) [reply src/elements/ChrUniformAcc.H]: fix via e1a4ff35b14609b28c535dd0ed8ffd19c2b9419f

File: src/elements/ChrUniformAcc.H [hunk 9]
```diff
@@ -169,8 +172,12 @@ namespace impactx::elements
                 T_Real const denom = -pti_tot + pzi_tot;
 
                 // compute focusing constant (1/m) and rotation angle (in rad)
-                T_Real const theta = m_alpha_iez * log(numer / denom);
+                T_Real const theta = (m_acc == 0_prt)? m_alpha * m_slice_ds / pzi_tot : m_alpha_iez * log(numer / denom);
                 auto const [sin_theta, cos_theta] = amrex::Math::sincos(theta);
+                T_Real const sin_theta_ialpha = m_alpha == 0_prt
+                    ? (m_acc == 0_prt ? m_slice_ds / pzi_tot
+                                      : log(numer / denom) / m_ez)
+                    : sin_theta / m_alpha;
```
Comment (ax3l) [inline src/elements/ChrUniformAcc.H:181]: If this nested branching is too tricky to read, then we can define another intermediate variable.

Comment (ax3l) [reply src/elements/ChrUniformAcc.H:181]: done

**Review by ax3l — APPROVED** [review]
